IaaS-Story/03-Static-public-IP/azuredeploy.json

```diff
@@ -0,0 +1,230 @@
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "vnetName": {
+      "type": "string",
+      "defaultValue": "WTestVNet",
+      "metadata": {
+        "description": "Name for the new VNet."
+      }
+    },
+    "vnetPrefix": {
+      "type": "string",
+      "defaultValue": "192.168.0.0/16",
+      "metadata": {
+        "description": "CIDR prefix for the VNet address space."
+      }
+    },
+    "frontEndSubnetName": {
+      "type": "string",
+      "defaultValue": "FrontEnd",
+      "metadata": {
+        "description": "Name for the front end subnet."
+      }
+    },
+    "frontEndSubnetPrefix": {
+      "type": "string",
+      "defaultValue": "192.168.1.0/24",
+      "metadata": {
+        "description": "CIDR address prefix for the front end subnet."
+      }
+    },
+    "stdStorageName": {
+      "type": "string",
+      "metadata": {
+        "description": "Name for standard storage account."
+      }
+    },
+    "stdStorageType": {
+      "type": "string",
+      "defaultValue": "Standard_LRS",
+      "allowedValues": [
+        "Standard_LRS",
+        "Standard_GRS",
+        "Standard_ZRS"
+      ],
+      "metadata": {
+        "description": "Storage type."
+      }
+    },
+    "osType": {
+      "type": "string",
+      "defaultValue": "Windows",
+      "allowedValues": [
+        "Windows",
+        "Ubuntu"
+      ],
+      "metadata": {
+        "description": "Type of OS to use for VMs: Windows or Ubuntu."
+      }
+    },
+    "adminUsername": {
+      "type": "string",
+      "metadata": { "description": "Username for local admin account." }
+    },
+    "adminPassword": {
+      "type": "securestring",
+      "metadata": { "description": "Password for local admin account." }
+    }
+  },
+  "variables": {
+    "webVMSettings": {
+      "Windows": {
+        "vmSize": "Standard_A1",
+        "publisher": "MicrosoftWindowsServer",
+        "offer": "WindowsServer",
+        "sku": "2012-R2-Datacenter",
+        "version": "latest",
+        "vmName": "WEB1",
+        "osdisk": "osdiskweb",
+        "nicName": "NICWEB1",
+        "pipName": "PIPWEB1",
+        "ipAddress": "192.168.1.101"
+      },
+      "Ubuntu": {
+        "vmSize": "Standard_A1",
+        "publisher": "Canonical",
+        "offer": "UbuntuServer",
+        "sku": "14.04.2-LTS",
+        "version": "latest",
+        "vmName": "WEB1",
+        "osdisk": "osdiskweb",
+        "nicName": "NICWEB1",
+        "pipName": "PIPWEB1",
+        "ipAddress": "192.168.1.101"
+      }
+    },
+    "location": "[resourceGroup().location]",
+    "vmStorageAccountContainerName": "vhds",
+    "vnetID": "[resourceId('Microsoft.Network/virtualNetworks',parameters('vnetName'))]",
+    "frontEndSubnetRef": "[concat(variables('vnetID'),'/subnets/',parameters('frontEndSubnetName'))]",
+    "webVMSetting": "[variables('webVMSettings')[parameters('osType')]]"
+  },
+    "resources": [
+      {
+        "apiVersion": "2015-05-01-preview",
```

Comment:
This can be 2015-06-15 but suggest making apiVersion a variable and reusing for resources